lib/provider/song_data_preprocess.dart

```diff
@@ -0,0 +1,67 @@
+class SongDataPreprocess {
```

Comment:
```suggestion
class SongDataPreprocessor {
```
클래스 이름이 위처럼 ~er 로 끝내는게 좋음
바꿀때 파일 이름도 `song_data_preprocessor`로  같이 변경 ㄱㄱ

또한 이 클래스는 `provider` 폴더보다 `service` 폴더 위치가 맞음